### PR TITLE
Runtime fix for ballistic firing sounds

### LINKED ISSUE #70332: obj/item/gun/ballistic/fire_sounds() runtime
https://github.com/tgstation/tgstation/blob/91f02f2a6b99c6eb5ae56fc3f7cfb903e703bc08/code/modules/projectiles/guns/ballistic.dm#L139-L140

div/null doesn't work.

### PR DESCRIPTION
**Pull request by lilBlovy**
## Bug fix for #70332

Its a bugfix for #70332

## Why It's Good For The Game

 Bug fix's very good

## Changelog

 Fixed a runtime bug.

:cl:
fix: fixed #70332
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
this is not a bugfix, max_ammo isn't ever supposed to be zero
fix why max_ammo is ending up as zero

**Comment by msgerbs** [comment]
> this is not a bugfix, max_ammo isn't ever supposed to be zero fix why max_ammo is ending up as zero

Isn't the issue that if there's no magazine then the whole thing returns a 0?

**Comment by GrimsonGrime** [comment]
Yes. It would show as zero with an empty mag. But now it doesnt show as zero. Even with an empty mag. And it doesnt change any other interactions with it so its fixed.

**Comment by Pickle-Coding** [comment]
This offset will change the frequency, and the difference is pretty significant for low magazine sizes such as shotguns. If receiving 0 magazine size from guns with no magazine is intended, then I recommend doing a ternary operator with the frequency, and setting it to 0 if magazine size is 0.

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -137,7 +137,7 @@
 	AddElement(/datum/element/weapon_description, attached_proc = .proc/add_notes_ballistic)
 
 /obj/item/gun/ballistic/fire_sounds()
-	var/frequency_to_use = sin((90/magazine?.max_ammo) * get_ammo())
+	var/frequency_to_use = sin((90/(magazine?.max_ammo+1)) * get_ammo())
```
Comment (Fikou) [inline code/modules/projectiles/guns/ballistic.dm:140]: make it be 1 if there's no max ammo instead

Comment (Fikou) [reply code/modules/projectiles/guns/ballistic.dm:140]: 90/(magazine?.max_ammo || 1)

Comment (msgerbs) [reply code/modules/projectiles/guns/ballistic.dm:140]: Isn't this what `max()` is for?

`sin(max(((90/(magazine?.max_ammo+1)) * get_ammo()), 90)`

**Comment by Mothblocks** [comment]
We have too many open PRs and you have week old unaddressed maintainer feedback, please contact Fikou on Discord when you are able to update your PR

**Pull request closed without merging**